### ISSUE TITLE
"TypeVar constraint type cannot be generic" false positive / incorrect error message when generic defaults reference each other

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1536]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnusedParameter=false&code=MYGwhgzhAEBiD28DaAVaBeaAhMAnAugFwB0pAsAFCWiQw66oZyJGnGWUAmApgGbS9EqQtjz4AFAEoRbaAGJoKAJ4AHbgDU80YPAB2EAC64wAS10HoB1d21hdu%2BBYBGNgObdd3XCeDRxubhV4XAMAcQ8vMBBlNQBJKABXbghJIA)

```python
class Foo[T = Bar]:...

class Bar[T = Foo]:...

def foo[T: Bar](): ... # TypeVar constraint type cannot be generic (reportGeneralTypeIssues)
```

it's unclear to me whether there should actually be an error here (i think probably yes) but the error message is wrong

**Comment by KotlinIsland** [comment]
i would have expected the `Bar` here to be `Bar[Any]`, unless i'm missing something

**Comment by DetachHead** [comment]
it would be `Bar[Foo[Bar[Foo[...]]]]`, so i guess that would turn into `Bar[Any]` because it would otherwise be an infinite recursion that can't be resolved?

**Comment by KotlinIsland** [comment]
oh, hmmmm, i did miss that 😅

we can't have infinitly recursive types? i thought they were supported?

```py
type L = list[L]
```
